nexus_motion_planner/src/test_motion_plan_cache.cpp

```diff
@@ -0,0 +1,1164 @@
+// Copyright 2023 Johnson & Johnson
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rclcpp/rclcpp.hpp>
+
+#include "moveit/robot_state/conversions.h"
+#include "moveit/robot_state/robot_state.h"
+#include "motion_plan_cache.hpp"
+
+#include <thread>
+
+using moveit::planning_interface::MoveGroupInterface;
+using nexus::motion_planner::MotionPlanCache;
+
+const std::string g_robot_name = "panda";
+const std::string g_robot_frame = "world";
+
+// UTILS =======================================================================
+// Utility function to emit a pass or fail statement.
+void check_and_emit(
+  bool predicate, const std::string& prefix, const std::string& label)
+{
+  if (predicate)
+  {
+    std::cout << "[PASS] " << prefix << ": " << label << std::endl;
+  }
+  else
+  {
+    std::cout << "[FAIL] " << prefix << ": " << label << std::endl;
```

Comment:
Consider exiting the program if there is a failure. Or have 2 helper functions, one to continue on failure and another to exit immediately.

Reply:
https://github.com/osrf/nexus/pull/28/commits/a06c57815925a4911231b86974f896b6e03928db